### DISCUSSION TITLE
Octolapse Install not working eventhough Plugins are installed

### DISCUSSION THREAD

**Discussion by JZiZu** [feelfreelinux/octo4a#370] [Q&A]
Hi! I am very thankfull this project was set up because of the prices of the RaspberryPis.
I hav now installed the new Version of Octo4a (1.2.0) on my old Samsung Galaxy A5 (2017) with Android 8.0.0 installed. 
The server itself is working right now and I installed the plugin extras but when I try to install Octolapse I am getting many errors and it fails installing. 
`error: command '/usr/bin/gcc' failed with exit code 1
[end of output]
note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure
× Encountered error while trying to install package.
╰─> Octolapse
note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
Fehler!
Could not parse output from pip, see plugin_pluginmanager_console.log for generated output
`
This is what I get at the end in the logs. 
Have anyone else faced this problem yet? Or am I missing something?

**Comment by lmichoux** [comment]
Hi, Just an idea
In the app, in the settings menu, check the option called "install plugin extras". Try that option and let us know if it helps.

Regards,

**Reply by lmichoux** [reply]
or your phone is scrwed or full of useless stuff. did you revert to factory default your old phone ?

**Reply by JZiZu** [reply]
Hi. Thanks for your reply.
Thats my problem. These extras are installed and my phone was reset to factory defaults before...
Any ideas?

**Comment by sv0o** [comment]
I believe I'm having the same issue while installing the Octolapse plugin.  I have o4a's plugin extras installed.  
The Remix box I'm using does have potentially 'useless stuff' installed but it doesn't have its own camera. (see my other thread)
The timelapse config in Octoprint is not getting a test snapshot to work as of yet, so maybe that is part of the problem, but I can't get to the point of tweaking Octolapse's camera settings because the install of the plugin keeps failing.

**Reply by lmichoux** [reply]
can you enable ssh, then ssh to your phone using any ssh client, and check the logs issuing the following commands: 
cd .octoprint/logs
ls -al
cat plugin_pluginmanager_console.log

About the end of the log file should get more info.
You can post here the result if this does not speak to you.

**Comment by sv0o** [comment]
I don't have to ssh.  I downloaded through the octoprint logging interface.  It's not speaking to me.

[plugin_pluginmanager_console.log](https://github.com/feelfreelinux/octo4a/files/10714405/plugin_pluginmanager_console.log)

**Comment by lmichoux** [comment]
Looks like you have a known issue.
see here for more details.
https://github.com/pypa/pip/issues/8559
You are missing some packages like "wheel". Without this you cannot install and compile stuff during installation.
I think the quick fix is to try another version of the octolapse package.

You can try this to update missing wheel package
# pip3 install --upgrade setuptools wheel tornado --no-cache-dir
Then rerun your install.

I hope this helps.
Regards,

**Reply by JZiZu** [reply]
Thanks for your reply! That seemed to do something but I guess something is still missing.


```
Installiere Plugin "Octolapse" von https://github.com/FormerLurker/Octolapse/archive/master.zip...
/usr/bin/python3 -m pip --disable-pip-version-check install file:///tmp/tmp307fdmix/Octolapse-master.zip --no-cache-dir
Processing /tmp/tmp307fdmix/Octolapse-master.zip
Preparing metadata (setup.py): started
error: subprocess-exited-with-error
× python setup.py egg_info did not run successfully.
│ exit code: 1
╰─> [10 lines of output]
Argon2 passlib backend is not available, not using it for password hashing
/tmp/pip-req-build-zvgish8u/setup.py:58: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
if LooseVersion(octoprint.server.VERSION) < LooseVersion("1.4"):
error in Octolapse setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Expected end or semicolon (after version specifier)
pillow >=6.2.0<7.0.0
~~~~~~~^
likely tags: v0.4.1
picking v0.4.1
got version from expanded keyword {'version': 'v0.4.1', 'full-revisionid': 'cec243c65a979e8bbe90f09fc5aec6d0844d2d3b', 'dirty': False, 'error': None, 'date': '2020-12-04T18:35:29-0600'}
Found packages: {'octoprint_octolapse_setuptools', 'octoprint_octolapse.test', 'octoprint_octolapse'}
[end of output]
note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
× Encountered error while generating package metadata.
╰─> See above for output.
note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Preparing metadata (setup.py): finished with status 'error'
Fehler!
Could not parse output from pip, see plugin_pluginmanager_console.log for generated output
```

**Reply by sv0o** [reply]
[plugin_pluginmanager_console.log](https://github.com/feelfreelinux/octo4a/files/10716383/plugin_pluginmanager_console.log)
I think that's what's happening to me as well.

**Comment by lmichoux** [comment]
On last thing. I am not much in code and dependencies, but It could be that you are installing a too recent dependency tree on a too old phone.
Have you tried latest v03 version of octolapse ? 
pip install https://github.com/FormerLurker/Octolapse/archive/v0.3.4.zip

**Reply by sv0o** [reply]
no I might try that.  At the moment the latest Octolapse runs great on my octo4A instance on an LG G4 phone.  It installed without any issues there and if I want a cool timelapse, I can go back to that.  I'm currently trying the dual octo4A setup I described in a different discussion for reliablity issues. I want to read through all of the differences between 0.3 and 0.4 releases to see if it is worth the time.  

but Octolapse hasn't been updated in more than two years.  It seems like it's overdue.

**Reply by JZiZu** [reply]
Now something interesting happened. It installed but after installing it can not be activated in the Puluginmanager because it is saying that it is not compatible with my Octoprint or Python Version...

**Comment by lmichoux** [comment]
Hi, 
I think I reproduced your issue. my last suggestion, to downgrade to lower version is not good, it is better to fix latest.
There are some changes to be done to get latest octolapse running on latest octo4a.
I found issues with gcc and python.
I managed to get it running on my old Lenny5 phone.
Here is a summary of my actions, I think everything is there : 
1st) to restart from clean / on the phone, clean octo4a APK memory and cache and uninstall / reinstall
2nd) install recommended plugins on your phone + enable SSH & start server
3rd) from ssh, unzip octolapse manually
wget wget https://github.com/eyal0/OctoPrint-PrintTimeGenius/archive/master.zip
unzip master.zip
cd Octolapse-master
pip3 install LooseVersion
pip3 install setuptools -U
here: edit setup.py
as LooseVersion is not working so here is the dirty fix.comment out lines : 59/60, 
#if LooseVersion(octoprint.server.VERSION) < LooseVersion("1.4"):
#    plugin_requires.extend(["flask_principal>=0.4,<1.0"])

again, in settup.py line 54/55 / this line syntax is not recognized, pillow is called Pillow but it does not work / outdated, replace :
#plugin_requires = ["Pillow >=6.2.0<7.0.0", "sarge", "six", "OctoPrint>1.3.8", "psutil", "file_read_backwards",
#                   "setuptools>=6.0", "awesome-slugify>=1.6.5,<1.7"]
by this small line (that is now recognised) :
plugin_requires = ["six"]
Install any missing package manually with pip3 
pip3 install file_read_backwards

to fix the error: converting to 'bool' from 'std::nullptr_t' requires direct-initialization [-fpermissive]
this is because GCC version is now too restrictive, edit octoprint_octolapse/data/lib/c/gcode_position_processor.cpp
replace return NULL;
by return false;

4) compile octolapse
cd Octolapse-master (is you are not already)
pip3 install .

5) restart octoprint / and now you should get octolapse menu not working... it is time to fix then fix settings issues because of python 3.10
see https://github.com/dropbox/PyHive/issues/392  for details. Importing ABC directly from collections has been removed in Python 3.10 in favor of collections.abc

so edit /usr/lib/python3.10/site-packages/octoprint_octolapse/settings.py
replace line 37
import collections
by 
from collections.abc import Iterable

then, in the rest of document replace ALL occurences of "collections.Iterable" by "Iterable"
exemple : if not isinstance(iterable, Iterable):

There are some other .py files with same issue, perform same change to make sure it is more stable.

6) restart octoprint / configure your printer & stuff. should work.

Tell me if OK.
I hope this will help the developers fixing this instance.

**Reply by JZiZu** [reply]
Its not working. I guess I made something wrong with this file : octoprint_octolapse/data/lib/c/gcode_position_processor.cpp
Do I have to replace every return NULL; with return false; ?

**Reply by lmichoux** [reply]
I updated only the one that was on line 1315

**Reply by lmichoux** [reply]
That change on line 1315 made the job, for the other lines, I think code should be analyzed in depth. because returning NULL means returning nothing, while returning false or true is returning something.

**Reply by JZiZu** [reply]
WOW! It finally works! 
Thank you very much!

**Reply by sv0o** [reply]
This is great news.  It sounds like the problem lies in Octolapse.  Should it be opened as an issue on https://github.com/FormerLurker/Octolapse?
I am very new to Github.  do people make issues that cross projects?

**Reply by JZiZu** [reply]
I am not sure. But I think its a good idea!

**Reply by lmichoux** [reply]
> WOW! It finally works! Thank you very much!

I am happy as you ! 
Happy timelapse!
https://www.youtube.com/shorts/WPwhdDWUcHM

**Reply by Jcub2491** [reply]
Is there a way to simplify these instructions any further? you're losing me at the third step.

**Reply by sv0o** [reply]
These instructions are meant for someone with experience in bash and busybox if not Alpine.  I had to figure out how to install vim for editing and then get some vim tips with google searches.  I stumbled through the third step and I might have done it right. After the fourth step, Octoprint shows the octolapse tab, but when I do the fifth step, octolapse disappears.  


After a couple of failed tries, I confirmed that he's saying to comment out two lines each time in setup.py

I don't know for certain if " compile octolapse" is its own step or if it describes  " install . " 

It's been a known issue for octolapse
See https://github.com/FormerLurker/Octolapse/issues/872

**Reply by rodrigo-ceccato** [reply]
Hey, did you mean wget octolapse?

**Reply by sv0o** [reply]
> Hey, did you mean wget octolapse?

don't bother, a-c-m rolled all the fixes into a fork of octolapse.  See https://github.com/feelfreelinux/octo4a/issues/373#issuecomment-1445310075

**Reply by rodrigo-ceccato** [reply]
Oh, sweet. Thanks
